src/parser/core/modules/CastTime.ts

```diff
@@ -155,6 +174,34 @@ export default class CastTime extends Analyser {
 	 * @returns The actual cast time, either as the default, or the modified time if any modifiers were in effect. Returns undefined if the action cannot be determined, or has no default cast time
 	 */
 	public forAction(actionId: number, timestamp: number = this.parser.currentEpochTimestamp): number | undefined {
+		return this.getAdjustedTime(actionId, timestamp)
+	}
+
+	/**
+	 * recastForAction Module-compatibility function
+	 * @deprecated
+	 */
+	public recastForFflogsAction(actionId: number, timestamp: number = this.parser.currentTimestamp): number | undefined {
+		return this.recastForAction(actionId, this.parser.fflogsToEpoch(timestamp))
+	}
+	/**
+	 * Returns the effective recast time for the specified action at the specified point in time
+	 * @param actionId The action in question
+	 * @param timestamp Thetimestamp in question
+	 * @returns The actual recast time, either as the default, or the modified time if any modifiers were in effect. Returns undefined if the action cannot be determined, or has no gcdRecast/cooldown property defined
+	 */
+	public recastForAction(actionId: number, timestamp: number = this.parser.currentEpochTimestamp): number | undefined {
+		return this.getAdjustedTime(actionId, timestamp, true)
+	}
+
+	/**
+	 * Returns the adjusted time (either cast or recast) for the specified action at the specified point in time
+	 * @param actionId The action in question
+	 * @param timestamp The timestamp in question
+	 * @param forRecast Do we want the recast for this action, or the cast time?
+	 * @returns The adjusted time, if any adjustments exist at this timestamp, or the default if not. Will return undefined if the base time (recast/cooldown/cast) can't be determined
+	 */
+	private getAdjustedTime(actionId: number, timestamp: number = this.parser.currentEpochTimestamp, forRecast: boolean = false): number | undefined {
```

Comment:
i was gonna say 'would be nicer to type this as any numeric key on action so we can use it for arbitrary stuff in the future', but, well... `type Foo = {[K in keyof Action]: number extends Action[K] ? K : never}[keyof Action]` gets the "notpretty" badge of meh from me.

I'd like to avoid an unnamed bool arg though - either swapping to a string union or preferably naming via an object

Reply:
Changed the 'which time does this adjustment affect?' tracking to be an explicit type, and updated the matchingTimes filter/function signatures to handle it. Tried to make it a transparent change for the callers, aside from Leylines needing to set the new affectsWhich = 'both' param. Did leave myself a todo note to transform the parameter vomit into object parameters in a separate PR, really don't want to balloon this one any farther than it already is 🙃